lib/Parse/ParseDecl.cpp

```diff
@@ -2739,6 +2739,23 @@ parseIdentifierDeclName(Parser &P, Identifier &Result, SourceLoc &Loc,
 
   P.checkForInputIncomplete();
 
+  if (P.Tok.is(tok::integer_literal) || P.Tok.is(tok::floating_literal) ||
+      (P.Tok.is(tok::unknown) && isdigit(P.Tok.getText()[0]))) {
+    // Per rdar://problem/32316666, using numbers for identifiers is a common
```

Comment:
Oh, I didn't realize that was a convention. You're thinking we should lean on `git blame` and also just put sufficient explanation into the code to generally not need to dig out the radar?